### PR TITLE
deno: fix clang build failure

### LINKED ISSUE #367962: Build failure: deno-2.1.4
## Steps To Reproduce

1. build deno-2.1.4

## Build log

<details>

<summary>Build Log</summary>

```
libtool: compile:  /nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -I. -I../include -Iinclude -I../src -Oz -ffunction-sect>
  make[2]: Leaving directory '/private/tmp/nix-build-deno-2.1.4.drv-1/source/target/aarch64-apple-darwin/release/build/libffi-sys-d47a2cef7411a0b3/out/libffi-build/aarch64-apple-darwin'
  make[1]: Leaving directory '/private/tmp/nix-build-deno-2.1.4.drv-1/source/target/aarch64-apple-darwin/release/build/libffi-sys-d47a2cef7411a0b3/out/libffi-build/aarch64-apple-darwin'

  --- stderr
  clang: error: unsupported option '-print-multi-os-directory'
  clang: error: no input files
  ../src/java_raw_api.c:328:46: warning: 'ffi_java_raw_size' is deprecated [-Wdeprecated-declarations]
    328 |   ffi_java_raw *raw = (ffi_java_raw*)alloca (ffi_java_raw_size (cif));
        |                                              ^
  include/ffi.h:326:56: note: 'ffi_java_raw_size' has been explicitly marked deprecated here
    326 | size_t ffi_java_raw_size (ffi_cif *cif) __attribute__((deprecated));
        |                                                        ^
  ../src/java_raw_api.c:331:3: warning: 'ffi_java_ptrarray_to_raw' is deprecated [-Wdeprecated-declarations]
    331 |   ffi_java_ptrarray_to_raw (cif, avalue, raw);
        |   ^
  include/ffi.h:322:93: note: 'ffi_java_ptrarray_to_raw' has been explicitly marked deprecated here
    322 | void ffi_java_ptrarray_to_raw (ffi_cif *cif, void **args, ffi_java_raw *raw) __attribute__((deprecated));
        |                                                                                             ^
  2 warnings generated.
  /private/tmp/nix-build-deno-2.1.4.drv-1/sysv-98dd48.s:27:2: error: invalid CFI advance_loc expression
   .cfi_def_cfa x1, 40;
   ^
  /private/tmp/nix-build-deno-2.1.4.drv-1/sysv-98dd48.s:188:2: error: invalid CFI advance_loc expression
   .cfi_adjust_cfa_offset (8*2 + (8 * 16 + 8 * 8) + 64)
   ^
  make[2]: *** [Makefile:1296: src/aarch64/sysv.lo] Error 1
  make[1]: *** [Makefile:1397: install-recursive] Error 1
  make: *** [Makefile:3158: install] Error 2
  thread 'main' panicked at /private/tmp/nix-build-deno-2.1.4.drv-1/deno-2.1.4-vendor.tar.gz/libffi-sys/build/common.rs:8:5:
```

</details>

## Metadata

- system: `"aarch64-darwin"`
 - host os: `Darwin 24.2.0, macOS 15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.25.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/73cnf2k7mji95fcflvay15pdw4skhhk5-source`
 
## Notify maintainers
@jk

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ofalvai**
https://hydra.nixos.org/build/282160517/log/tail

Fixes #367962

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ofalvai** [comment]
@ofborg build deno.tests

File: pkgs/by-name/de/deno/package.nix
Comment (paparodeo) [inline pkgs/by-name/de/deno/package.nix:64]:
```suggestion
  buildInputs = [ libffi ];
```

both x64 and aarch64 are now using the same sdk (11.3) so this is no longer needed. -- oh, but I was probably just the formatter that made these lines looked changed...

Comment (ofalvai) [reply pkgs/by-name/de/deno/package.nix:64]: I can remove the SDK package, but I'm not sure about the min version hook (I'm not sure I fully understand the implications of the deployment target setting).

I looked it up in the stdenv docs, and it says the following:

> On macOS, the `darwinMinVersion` and `darwinSdkVersion` are always the same, and are currently set to 11.3.

So removing the hook would _increase_ the minimum supported version to 11.3, right?

Comment (paparodeo) [reply pkgs/by-name/de/deno/package.nix:64]: it can be removed. it was probably just added for x64 darwin to bump it the least amount to get things to compile. but everything is at 11 now so I can't see the point in lowering it.

**Comment by ofalvai** [comment]
@ofborg build deno.tests

File: pkgs/by-name/de/deno/package.nix
Comment (ofalvai) [reply pkgs/by-name/de/deno/package.nix]: Done, I think it's ready for merge now

**Review by paparodeo — APPROVED** [review]
deno.tests  builds fine for me on x64 darwin

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368102`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>windmill</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>windmill</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
Thanks

**Comment by 06kellyjac** [comment]
@khaneliman @ofalvai I would appreciate being CCed and being given more than 1 day to review the changes and consider the addition of a new maintainer, especially over the holidays.
I don't think new maintainer should be added without the input of (active) maintainers.

The commit to add a maintainer should have been prefixed `deno:` as it was relevant to the deno package, not to the maintainers list.

Also a formatting commit which resolves incorrect formatting from a commit added in the same PR should be squashed together please.

**Comment by ofalvai** [comment]
@06kellyjac I added myself as a maintainer with the assumption that the more maintainers the better, and that most packages in Nix lack active maintainers. In this specific case, the last staging-next iteration broke many packages including Deno. Looks like I was (one of) the first to notice Deno, so I decided to investigate it. And because I'm using the Deno package personally, I want to keep it working and I'm happy to help in the future too. I don't understand your criticism, but I could be missing something.

**Comment by 06kellyjac** [comment]
Anyone is welcome to submit PRs fixing issues in a package. CC-ing maintainers is advisable as maintainers being attached as reviewers can be slow if ofborg is having a slow day. Also requests for review don't show in GH notifications.

I welcome more maintainers. There are several nixpkgs contributors who frequently help out with the deno package, who aren't down in the package meta as maintainers, but would also be great additions. There are several active packages I've joined the maintenance of and several inactive or fully abandoned packages I've adopted.
It's just courteous to CC the existing maintainers and get at least one approval.
If you've identified there aren't any who are active then fair enough.

I'd be interested to see some numbers on packages with inactive maintainership if you have them to hand. Abandoned ones with no maintainers listed are a bit easier to find.

Kind regards,
Jack

**Pull request merged**
